active/AnimatedIcon/AnimatedIconSpec.md

```diff
@@ -0,0 +1,282 @@
+
+
+
+# API Pages
+
+## AnimatedIcon class
+
+An AnimatedIcon is a UI component which plays animated images in response to user interaction and visual state changes.
+
+On Windows, the VisualStateManager is used to manage the logic for transitioning between states for controls.
+Each control has a specified VisualStateGroup, which contains mutually exclusive VisualState objects.
+For example, a Button may have a VisualStateGroup called “CommonStates”, which specifies a VisualState object for
+“Normal” “Pressed”, “PointerOver”, and “Disabled”.
+
+A control will use the VisualStateManager to determine what state the animated icon is transitioning to and
+play the corresponding animation segment of the icon.
+The icon code will need to have markers named in a standard way (described below) so
+AnimatedIcon can map the correct animation segment with the visual state the control is transitioning to.  
+
+**Important APIs:** [AnimatedIcon class](https://docs.microsoft.com/uwp/api/microsoft.ui.xaml.controls.AnimatedIcon)
+
+### Is this the right control? 
+
+Use an AnimatedIcon control when you want to tie an animated icon to a user interaction in your UI.
+Do not use an AnimatedIcon control if you want to just play an animation once or control the playback of the animation.
+Do not use an AnimatedIcon control if you want to play your animation in a control that does not support IconElement.
+Instead, use the AnimatedVisualPlayer control. 
+
+An AnimatedIcon control can be used anywhere an IconElement can be used in WinUI. 
+
+## How is this different than [AnimatedVisualPlayer](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.AnimatedVisualPlayer)?
+
+AnimatedIcon is an
+[IconElement](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.IconElement),
+and so can be used anywhere an element or IconElement is required, such as
+[NavigationViewItem.Icon](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.NavigationViewItem.Icon).
+AnimatedVisualPlayer is a more  general animation player that can be used anywhere in an application where AnimatedIcon is intended to only be used where icons are used in WinUI. 
+
+AVP allows developers to control animations using the following methods: Pause, PlayAsync, Resume, SetProgress, and Stop. When using AVP, developers may need to write a lot of code when getting their generated animation to properly respond to user interaction and state changes. This is the main distinction between the workflows for AVP and AnimatedIcon. An AnimatedIcon will be used when animations need to be driven by state changes and user interaction in a markup friendly way. It would reduce, or ideally eliminate the amount of code-behind needed to implement such scenarios.
+
+# Examples
+
+## Swap out a static icon with an animated icon in a Navigation View Item using the standard markers
+
+An icon animation will be activated by changes of state in XAML input controls (e.g. Button, and ToggleSwitch).
```

Comment:
The examples here show the case of an Icon property (like NavigationViewItem), how does it work with Button, which doesn't have an Icon property?

Reply:
If the default style for button gets updated (not currently planned) we could support this just working (with just the basic interaction states):
```
<Button>
    <AnimatedIcon>
        <MyAnimatedVisual/>
    </AnimatedIcon>
</Button>
```

Without an update to the default button style, you would need to either 1) retemplate button to set the animatedIcon.state attached properties on the content presenter in button. or 2) set the state property from code behind.

Reply:
Yeah since we are not planning on updating button for V1 I will remove that from the spec so it's not confusing.

Reply:
Anything you want to keep is fine to put into the Appendix so that we can find it later.

Reply:
updated and removed the button scenario.